<?xml version="1.0" ?>

<launch>
  <arg name="camera_name" default="camera_L"/>
  <arg name="frame_id" default="soma_link"/>
  <arg name="input_points" default="/$(arg camera_name)/prefiltered"/>
  <!-- <arg name="input_points" default="/$(arg camera_name)/depth_registered/points"/> -->
  <arg name="input_imu" default="/imu/data"/>

  <!-- <arg name="file_name" value="/home/soma/10_0.05_6.pcd"/> -->
  <arg name="file_name" value="/home/soma/Documents/tomokawa/pcd/Mirais_map_raw.pcd"/>
  <!-- <arg name="file_name" value="/home/soma/Documents/tomokawa/dataset/20210120_right/raw/1611116571.263151804.pcd"/> -->
  <arg name="rate" value="0.1"/>
  <node pkg="pcl_ros" type="pcd_to_pointcloud" name="pcd_to_pointcloud" args="$(arg file_name) $(arg rate)" output="screen">
      <param name="frame_id" value="map" />
  </node>

  <!-- <node pkg="pcl_ros" type="convert_unorganized_to_organized" name="convert_organized" args="$(arg file_name) $(arg rate)" output="screen">
    <remap from="~input" to="cloud_pcd" />
    <remap from="~output" to="organized_pcd"/>
  </node> -->
</launch>